sql/core/src/main/java/org/apache/spark/sql/sources/v2/Table.java

```diff
@@ -27,11 +28,11 @@
  * implementation can be a directory on the file system, a topic of Kafka, or a table in the
  * catalog, etc.
  * <p>
- * This interface can mixin the following interfaces to support different operations, like
- * {@code SupportsRead}.
+ * This interface can mixin {@link SupportsRead} and {@link SupportsWrite} to provide data reading
+ * and writing ability.
  */
 @Evolving
-public interface Table {
+public interface Table extends BaseStreamingSink {
```

Comment:
Why did this add `BaseStreamingSink`?

Reply:
`BaseStreamingSink` is an empty internal interface, which is a common interface for v1 and v2 streaming sources. We will remove it after ds v2 is finalized and v1 streaming source can be removed.

If we don't want to pollute the `Table` interface, we can create a
```
public interface StreamingTable extends Table, BaseStreamingSink
```

Reply:
@cloud-fan, your reply is not a justification for adding this. Please answer: **Why is this change required for this commit to function correctly?**

Reply:
It's required because it's the common interface for streaming in V1 and V2. A streaming sink is represented by a Sink in V1 or a Table in V2; BaseStreamingSink is used in places where it doesn't matter which kind of sink it is.

Reply:
Because `Table` is used to identify a source in `MicroBatchExecution`, to track the progress of all the sources in a micro-batch query.

This is only a problem in `MicroBatchExecution`, because it needs to support micro-bath source v1 (there is no continuous source v1) and ds v2 together. The way we did it is to create a common interface as the source representation for both micro-bath source v1 and ds v2, so that we can unify the code to track progress of sources, see https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

Reply:
Why can't implementations extend `BaseStreamingSink` and leave this interface alone? Not all tables are streaming sinks, so this inheritance is incorrect.